lib/takeaway.rb

```diff
@@ -0,0 +1,64 @@
+require 'twilio-ruby'
+require_relative 'menu'
+require_relative 'sms'
+
+class Takeaway
+
+
+  attr_reader :menu, :order
+
+  def initialize(menu = Menu.new)
+    @menu = menu
```

Comment:
You mention struggling with dependency injection in your reflection, but this is a perfect example of the pattern.

Essentially what you're doing here is saying 'when I create a Takeaway object, it should take one parameter called menu.  Whatever I pass in here, will then be stored in an instance variable called menu.  If I don't pass anything in, this parameter should default to a newly initialised Menu object.'

This gives you greater flexibility in what you pass into your Menu object, and also makes isolating your unit tests possible.